### DISCUSSION TITLE
Rhino Engine as Part of other Products

### DISCUSSION THREAD

**Discussion by StSchnell** [mozilla/rhino#1425] [General]
When I started working in data center automation, I got to know the Rhino JavaScript engine as part of VMware Aria Automation. So I asked myself in which products the Rhino engine is still used everywhere. Here is a tiny list:

- VMware Aria Automation 8.18.1 [Source](https://techdocs.broadcom.com/us/en/vmware-cis/aria/aria-automation/8-18/vco-installing-and-configuring-8-18/key-features-of-the-orchestrator-platform.html) and VCF Automation 9.0.1 [Source](https://techdocs.broadcom.com/us/en/vmware-cis/vcf/vcf-9-0-and-later/9-0/configuration-of-vmware-cloud-foundation-operations-orchestrator/key-features-of-the-vcf-operations-orchestrator-platform.html) for data center automation
- ThingWorx 9.7 Industrial IoT-Plattform, [Source](https://support.ptc.com/help/thingworx/platform/r9.7/en/index.html#page/ThingWorx/Help/Composer/Things/ThingServices/RhinoJavaScriptEngine.html)
- Apigee Google Cloud’s native API management tool to build, manage, and secure APIs, [Source](https://cloud.google.com/apigee/docs/api-platform/reference/policies/javascript-policy)
- Mirth Connect 4.5 integration engine to natively support and integrate with the most common system and healthcare standards.
- IBM MobileFirst Platform Foundation on IBM Containers until release 7.1.0, an integrated platform to extend the business to mobile devices.
- IBM Rational Software Architect 9.7.0, [Source](https://www.ibm.com/docs/en/rational-soft-arch/9.7.0?topic=launching-rhino-launch-shortcut)
- IBM Tivoli Netcool/Impact 7.1.0 is a set of runnable server components that work together to provide event management and integration functionality, [Source](https://www.ibm.com/docs/en/tivoli-netcoolimpact/7.1.1?topic=fundamentals-setting-javascript-version-use)
- Memento Database 3.4.0 to track and organize anything with a customizable database, [Source](https://wiki.mementodatabase.com/index.php/JavaScript_field)
- RingoJS 4.0.0 JavaScript platform built on the JVM and optimized for server-side applications, [Source](https://ringojs.org/)
- Adobe Experience Manager (AEM) 6.5 is a comprehensive content management solution that makes it easy to manage your marketing content and assets, [Source](https://developer.adobe.com/experience-manager/reference-materials/6-5/javadoc/overview-summary.html)
- Apache Druid 31.0.1, a real-time analytics database designed for fast slice-and-dice analytics on large data sets, [Source](https://druid.apache.org/docs/latest/development/javascript/)
- ServiceNow an intelligent platform for digital transformation, [Source](https://hishowcase.service-now.com/kb?id=kb_article_view&sysparm_article=KB0721865)
- BiRT (Business Intelligence Reporting Tool) 4.18.0 to create data visualizations, dashboards and reports that can be embedded into web applications and rich clients. [Source](https://download.eclipse.org/birt/updates/release/latest/) (Download birt-runtime and search for org.mozilla.rhino*)
- LibreOffice, a free office suite, and a successor to OpenOffice. [Source](https://wiki.documentfoundation.org/Documentation/DevGuide/Scripting_Framework)
- Sahi Pro 11.0.1, test automation tool providing automation across browser, desktop, mobile & SAP applications. [Source](https://resources.sahipro.com/docs/scripting/calling-java.html)

These fiveteen examples show us where the Rhino engine is used, and there are probably more examples.
If there are any other examples I would be happy to add to this list.

**Comment by EricTerrell** [comment]
I used it for my programmable calculator:

https://play.google.com/store/apps/details?id=com.ericbt.rpncalcpaid
https://github.com/EricTerrell/EBTCalc.Android

**Comment by zozoyasuda** [comment]
https://wiki.servoy.com/display/DOCS/Implementing+Business+Logic
It's used in Servoy Developer

**Reply by StSchnell** [reply]
New Link for Servoy 2024.06: [https://docs.servoy.com/release-notes/release-notes/2024.06#lib-upgrades](https://docs.servoy.com/release-notes/release-notes/2024.06#lib-upgrades)

**Comment by CardContact** [comment]
We use Rhino in the Smart Card Shell and ScriptingServer of [OpenSCDP](https://www.openscdp.org).

**Comment by tonygermano** [comment]
You can't leave https://github.com/HtmlUnit/htmlunit off the list. The main developer from that project is also a top contributor to Rhino.

**Comment by rbri** [comment]
Spend some minutes to download the dependants of the project (using HtmlUnit :-)).
Now we are able to sort them by stars to get an idea of the well known projects using Rhino.

<img width="733" height="1249" alt="grafik" src="https://github.com/user-attachments/assets/bb35e4e5-bed9-498f-8436-bef2433aa862" />

[rhino-stats3.txt](https://github.com/user-attachments/files/22602440/rhino-stats3.txt)

**Reply by StSchnell** [reply]
Thank you very much Ronald [@rbri], very interesting. 👍
Another good source is the [usages of the Maven Repository](https://mvnrepository.com/artifact/org.mozilla/rhino/usages).
All these sources make it clear how widely Rhino is used.

**Comment by aardvark179** [comment]
Could you do that as a table rather than a screenshot?

**Reply by rbri** [reply]
😀 the attached file is a csv file...

**Reply by aardvark179** [reply]
Yeah. I was on my phone so my choice was zooming into a screenshot or trying to read a word wrapped csv. ☹️

**Comment by michbarsinai** [comment]
* Used in BPjs, a Java library for creating formal models of system behavior https://github.com/bThink-BGU/BPjs

**Comment by rbri** [comment]
@michbarsinai it will be great if you can provide a PR for https://github.com/mozilla/rhino/blob/master/USAGE.md

**Reply by michbarsinai** [reply]
done
